more -Wunused-but-set-variable warnings

The following shows up in CI emscripten build jobs:
```
src/loaders/st_load.c:211:6: warning: variable 'smp_size' set but not used [-Wunused-but-set-variable]
        int smp_size;
            ^

src/loaders/ims_load.c:150:9: warning: variable 'smp_size' set but not used [-Wunused-but-set-variable]
    int smp_size;
        ^

src/loaders/prowizard/p61a.c:36:9: warning: variable 'ssize' set but not used [-Wunused-but-set-variable]
    int ssize = 0;
        ^
src/loaders/prowizard/p61a.c:566:9: warning: variable 'ssize' set but not used [-Wunused-but-set-variable]
    int ssize;
        ^

src/mix_paula.c:133:1: warning: parameter 'vr' set but not used [-Wunused-but-set-parameter]
MIXER(mono_a500)
^
src/mixer.h:18:25: note: expanded from macro 'MIXER'
        int count, int vl, int vr, int step, int ramp, int delta_l, int delta_r)
                               ^
src/mix_paula.c:140:1: warning: parameter 'vr' set but not used [-Wunused-but-set-parameter]
MIXER(mono_a500_filter)
^
src/mixer.h:18:25: note: expanded from macro 'MIXER'
        int count, int vl, int vr, int step, int ramp, int delta_l, int delta_r)
                               ^

src/depackers/uncompress.c:74:7: warning: variable 'bytes_in' set but not used [-Wunused-but-set-variable]
        long bytes_in;                  /* Total number of byte from input */
             ^
```

We also have this:
```
src/depackers/bunzip2.c: In function 'get_next_block':
src/depackers/bunzip2.c:129:21: warning: variable 'hufGroup' might be clobbered by 'longjmp' or 'vfork' [-Wclobbered]
  struct group_data *hufGroup;
                     ^
```

> ```
> src/loaders/st_load.c:211:6: warning: variable 'smp_size' set but not used [-Wunused-but-set-variable]
>         int smp_size;
>             ^
> 
> src/loaders/ims_load.c:150:9: warning: variable 'smp_size' set but not used [-Wunused-but-set-variable]
>     int smp_size;
>         ^
> 
> src/loaders/prowizard/p61a.c:36:9: warning: variable 'ssize' set but not used [-Wunused-but-set-variable]
>     int ssize = 0;
>         ^
> src/loaders/prowizard/p61a.c:566:9: warning: variable 'ssize' set but not used [-Wunused-but-set-variable]
>     int ssize;
>         ^

These are more remnants of code that was either moved to test functions or ProWizard code that is no longer relevant, easy fixes.

>```
> src/mix_paula.c:133:1: warning: parameter 'vr' set but not used [-Wunused-but-set-parameter]
> MIXER(mono_a500)
> ^
> src/mixer.h:18:25: note: expanded from macro 'MIXER'
>         int count, int vl, int vr, int step, int ramp, int delta_l, int delta_r)
>                                ^
> src/mix_paula.c:140:1: warning: parameter 'vr' set but not used [-Wunused-but-set-parameter]
> MIXER(mono_a500_filter)
> ^
> src/mixer.h:18:25: note: expanded from macro 'MIXER'
>         int count, int vl, int vr, int step, int ramp, int delta_l, int delta_r)
>                                ^
>```

More useless warnings a la `-Wunused-parameter`, ew.

>```
> src/depackers/uncompress.c:74:7: warning: variable 'bytes_in' set but not used [-Wunused-but-set-variable]
>         long bytes_in;                  /* Total number of byte from input */
>              ^
> ```

Not sure what's up with this one, but it can be commented out same as `bytes_out`.

> ```
> src/depackers/bunzip2.c: In function 'get_next_block':
> src/depackers/bunzip2.c:129:21: warning: variable 'hufGroup' might be clobbered by 'longjmp' or 'vfork' [-Wclobbered]
>   struct group_data *hufGroup;
>                      ^
> ```

"Checking for an EOF state is hard so we're going to use `setjmp`/`longjmp` pseudo-exceptions! We don't care that the fix for potentially clobbered pointers is making them `volatile`, which makes our code slower in the long run!"

--- a/src/depackers/bunzip2.c
+++ b/src/depackers/bunzip2.c
@@ -132,7 +132,7 @@ static int get_next_block(bunzip_data *bd)
 	unsigned char uc, symToByte[256], mtfSymbol[256], *selectors;
 	unsigned int *dbuf,origPtr;
 
-	hufGroup=NULL;
+	hufGroup=bd->groups; /* silence -Wclobbered */
 	dbuf=bd->dbuf;
 	dbufSize=bd->dbufSize;
 	selectors=bd->selectors;

--- a/src/depackers/bunzip2.c
+++ b/src/depackers/bunzip2.c
@@ -132,13 +132,15 @@ static int get_next_block(bunzip_data *bd)
 	unsigned char uc, symToByte[256], mtfSymbol[256], *selectors;
 	unsigned int *dbuf,origPtr;
 
-	hufGroup=bd->groups; /* silence -Wclobbered */
-	dbuf=bd->dbuf;
-	dbufSize=bd->dbufSize;
-	selectors=bd->selectors;
 	/* Reset longjmp I/O error handling */
 	i=setjmp(bd->jmpbuf);
 	if(i) return i;
+
+	hufGroup=NULL;
+	dbuf=bd->dbuf;
+	dbufSize=bd->dbufSize;
+	selectors=bd->selectors;
+
 	/* Read in header signature and CRC, then validate signature.
 	   (last block signature means CRC is for whole file, return now) */
 	i = get_bits(bd,24);

--- a/src/depackers/uncompress.c
+++ b/src/depackers/uncompress.c
@@ -71,7 +71,7 @@ static int decrunch_compress(FILE * in, FILE * out, long inlen)
 	int block_mode;
 	int i;
 
-	long bytes_in;			/* Total number of byte from input */
+	/*long bytes_in;*/		/* Total number of byte from input */
 	/*long bytes_out;*/		/* Total number of byte to output */
 	char_type inbuf[IBUFSIZ + 64];	/* Input buffer */
 	char_type outbuf[OBUFSIZ + 2048];/* Output buffer */
@@ -98,7 +98,7 @@ static int decrunch_compress(FILE * in, FILE * out, long inlen)
 		return -1;
 	}
 
-	bytes_in = insize;
+	/*bytes_in = insize;*/
 	maxcode = MAXCODE(n_bits = INIT_BITS) - 1;
 	bitmask = (1 << n_bits) - 1;
 	oldcode = -1;
@@ -255,7 +255,7 @@ static int decrunch_compress(FILE * in, FILE * out, long inlen)
 			oldcode = incode;	/* Remember previous code.      */
 		}
 
-		bytes_in += rsize;
+		/* bytes_in += rsize; */
 	}
 	while (rsize > 0);
 

--- a/src/loaders/ims_load.c
+++ b/src/loaders/ims_load.c
@@ -147,7 +147,6 @@ static int ims_load(struct module_data *m, HIO_HANDLE *f, const int start)
 {
     struct xmp_module *mod = &m->mod;
     int i, j;
-    int smp_size;
     struct xmp_event *event;
     struct ims_header ih;
     uint8 ims_event[3];
@@ -158,7 +157,6 @@ static int ims_load(struct module_data *m, HIO_HANDLE *f, const int start)
     mod->chn = 4;
     mod->ins = 31;
     mod->smp = mod->ins;
-    smp_size = 0;
 
     hio_read (ih.title, 20, 1, f);
 
@@ -170,8 +168,6 @@ static int ims_load(struct module_data *m, HIO_HANDLE *f, const int start)
 	ih.ins[i].volume = hio_read8(f);
 	ih.ins[i].loop_start = hio_read16b(f);
 	ih.ins[i].loop_size = hio_read16b(f);
-
-	smp_size += ih.ins[i].size * 2;
     }
 
     ih.len = hio_read8(f);

--- a/src/loaders/prowizard/p40.c
+++ b/src/loaders/prowizard/p40.c
@@ -63,7 +63,7 @@ static int depack_p4x(HIO_HANDLE *in, FILE *out)
 	uint16 track_addr[128][4];
 	long in_size;
 	int trkdat_ofs, trktab_ofs, smp_ofs;
-	int ssize = 0;
+	/* int ssize = 0; */
 	int SampleAddress[31];
 	int SampleSize[31];
 	int i, j, k, l, a, b, c;
@@ -125,7 +125,7 @@ static int depack_p4x(HIO_HANDLE *in, FILE *out)
 		SampleAddress[i] = ins.addr;
 		ins.size = hio_read16b(in);		/* sample size */
 		SampleSize[i] = ins.size * 2;
-		ssize += SampleSize[i];
+		/* ssize += SampleSize[i]; */
 		ins.loop_addr = hio_read32b(in);	/* loop start */
 		ins.loop_size = hio_read16b(in);	/* loop size */
 		ins.fine = 0;

--- a/src/loaders/prowizard/p61a.c
+++ b/src/loaders/prowizard/p61a.c
@@ -33,7 +33,7 @@ static int depack_p61a(HIO_HANDLE *in, FILE *out)
     int taddr[128][4];
     int tdata_addr = 0;
     int sdata_addr = 0;
-    int ssize = 0;
+    /* int ssize = 0; */
     int i = 0, j, k, l, a, b, z;
     int smp_size[31];
     int saddr[31];
@@ -95,7 +95,7 @@ static int depack_p61a(HIO_HANDLE *in, FILE *out)
 	        saddr[i] = saddr[i - 1] + smp_size[i - 1];
             }
 	    smp_size[i] = j * 2;
-	    ssize += smp_size[i];
+	    /* ssize += smp_size[i]; */
 	}
 	j = smp_size[i] / 2;
 	write16b(out, isize[i]);
@@ -563,7 +563,7 @@ static int test_p61a(const uint8 *data, char *t, int s)
     int nins;
     int pattern_data_offset;
     int sample_data_offset;
-    int ssize;
+    /* int ssize; */
 
 #if 0
     if (i < 7) {
@@ -598,14 +598,16 @@ static int test_p61a(const uint8 *data, char *t, int s)
     }
 
     /* test sample sizes and loop start */
-    ssize = 0;
+    /* ssize = 0; */
     for (i = 0; i < nins; i++) {
 	len = readmem16b(data + i * 6 + 4);
 	if ((len <= 0xffdf && len > 0x8000) || len == 0)
 	    return -1;
 
+	/*
 	if (len < 0xff00)
 	    ssize += len * 2;
+	*/
 
 	lstart = readmem16b(data + i * 6 + 8);
 	if (lstart != 0xffff && lstart >= len)

--- a/src/loaders/prowizard/theplayer.c
+++ b/src/loaders/prowizard/theplayer.c
@@ -196,7 +196,7 @@ static int theplayer_depack(HIO_HANDLE *in, FILE *out, int version)
     /*uint8 pack = 0;*/
     int taddr[128][4];
     int sdata_addr = 0;
-    int ssize = 0;
+    /* int ssize = 0; */
     int i, j, k;
     int smp_size[31];
     int saddr[31];
@@ -272,7 +272,7 @@ static int theplayer_depack(HIO_HANDLE *in, FILE *out, int version)
 	        saddr[i] = saddr[i - 1] + smp_size[i - 1];
             }
 	    smp_size[i] = j * 2;
-	    ssize += smp_size[i];
+	    /*ssize += smp_size[i];*/
 	}
 	j = smp_size[i] / 2;
 

--- a/src/loaders/st_load.c
+++ b/src/loaders/st_load.c
@@ -208,7 +208,6 @@ static int st_load(struct module_data *m, HIO_HANDLE *f, const int start)
 {
 	struct xmp_module *mod = &m->mod;
 	int i, j;
-	int smp_size;
 	struct xmp_event ev, *event;
 	struct st_header mh;
 	uint8 mod_event[4];
@@ -224,7 +223,6 @@ static int st_load(struct module_data *m, HIO_HANDLE *f, const int start)
 	mod->chn = 4;
 	mod->ins = 15;
 	mod->smp = mod->ins;
-	smp_size = 0;
 
 	hio_read(mh.name, 1, 20, f);
 	for (i = 0; i < 15; i++) {
@@ -274,8 +272,6 @@ static int st_load(struct module_data *m, HIO_HANDLE *f, const int start)
 		    || mh.ins[i].loop_size > 0x1387) {
 			ust = 0;
 		}
-
-		smp_size += 2 * mh.ins[i].size;
 	}
 
 	if (libxmp_init_instrument(m) < 0) {

--- a/src/mix_paula.c
+++ b/src/mix_paula.c
@@ -125,6 +125,10 @@ static void do_clock(struct paula_state *paula, int cycles)
     unsigned int pos = vi->pos; \
     int frac = (1 << SMIX_SHIFT) * (vi->pos - (int)vi->pos)
 
+#define VAR_PAULA_MONO(x) \
+    VAR_NORM(x); \
+    vl <<= 8
+
 #define VAR_PAULA(x) \
     VAR_NORM(x); \
     vl <<= 8; \
@@ -132,14 +136,14 @@ static void do_clock(struct paula_state *paula, int cycles)
 
 MIXER(mono_a500)
 {
-	VAR_PAULA(int8);
+	VAR_PAULA_MONO(int8);
 
 	LOOP { PAULA_SIMULATION(0); MIX_MONO(); }
 }
 
 MIXER(mono_a500_filter)
 {
-	VAR_PAULA(int8);
+	VAR_PAULA_MONO(int8);
 
 	LOOP { PAULA_SIMULATION(1); MIX_MONO(); }
 }

